pandas/core/groupby/generic.py

```diff
@@ -967,9 +967,9 @@ def aggregate(self, func=None, *args, **kwargs):
                         raise
                     result = self._aggregate_frame(func)
                 else:
-                    result.columns = Index(
-                        result.columns.levels[0], name=self._selected_obj.columns.name
-                    )
+                    result.columns = result.columns.rename(
+                        [self._selected_obj.columns.name] * result.columns.nlevels
+                    ).droplevel(-1)
```

Comment:
can you add a comment about what is going on here, GH ref back to the issue/PR

Reply:
@jbrockmendel sure, added. Anything else that needs doing here?